### DISCUSSION TITLE
Black Screen while install Kali Linux on Macbook Air M1

### DISCUSSION THREAD

**Discussion by ljis120301** [utmapp/UTM#4044] [UTM Help (macOS)] [answered]
### Discussed in https://github.com/utmapp/UTM/discussions/4035

<div type='discussions-op-text'>

<sup>Originally posted by **theanuraaag** May 17, 2022</sup>
I am getting a black screen when I tried to install Kali Linux on my Macbook Air M1 using UTM, it works normally till the option install, graphical install, etc but when I click on Graphical Install a black screen appears and nothing happens after that.
<img width="1440" alt="Screenshot 2022-05-18 at 11 31 28 AM" src="https://user-images.githubusercontent.com/86941924/168968310-7d79bd34-59c1-4f56-9c0a-6e8cf80633b8.png">
<img width="1440" alt="Screenshot 2022-05-18 at 11 31 36 AM" src="https://user-images.githubusercontent.com/86941924/168968320-da2b90da-dfc2-4c38-bf39-a6fdfca4c5cb.png">
<img width="1440" alt="Screenshot 2022-05-18 at 11 32 51 AM" src="https://user-images.githubusercontent.com/86941924/168968345-11758afd-751f-4507-a388-f08e7b096255.png">

</div>

**Comment by DylKhn** [comment, marked as answer]
If you are using Kali 2022.2 use the older version 2022.1 this fixed it for me. Seems to be a problem with the latest release.

**Reply by ghost** [reply]
Where would I download the older version? I tried searching for it but I couldn't find any leads Cheers!

**Reply by shivq43** [reply]
you can download that from http://old.kali.org/kali-images/kali-2022.1/

**Reply by DrewBrasher** [reply]
Thanks! I had the same issue and version 2022.1 arm64 worked for me also.

**Reply by Himzzii** [reply]
> If you are using Kali 2022.2 use the older version 2022.1 this fixed it for me. Seems to be a problem with the latest release.

can you please share the specific link as i am using mac air m1 and i am not able to find the perfect kali on this link

**Reply by turborg2011** [reply]
> > If you are using Kali 2022.2 use the older version 2022.1 this fixed it for me. Seems to be a problem with the latest release.
> 
> can you please share the specific link as i am using mac air m1 and i am not able to find the perfect kali on this link

I think you need this link - it is installer for arm: http://old.kali.org/kali-images/kali-2022.1/kali-linux-2022.1-installer-arm64.iso

**Reply by L0n35** [reply]
Thanks! 2022.1 worked for me. 2022.2 and 2022.3 didnt.

**Reply by asif-kabir-emon** [reply]
Thanks, it also works for me.

**Reply by dkbast** [reply]
same here

**Reply by TincoNomad** [reply]
Same, gracias

**Reply by PandaPando** [reply]
Thanks man !

**Reply by goelarnav12** [reply]
thanks a lot......

**Reply by pyxl-dev** [reply]
Thanks !

**Comment by eyyupbarut** [comment]
You can go by first option "Install", I will work. After configuration (its like Graphical Install but you just cant use mouse) you will be able to run Kali.

**Reply by ghost** [reply]
It prompts me on the installation screen, any method to fix this?

**Reply by barvhaim** [reply]
it shows the same black screen

**Reply by jcdevsoft** [reply]
<img width="792" alt="Captura de pantalla 2023-01-04 a la(s) 22 53 29" src="https://user-images.githubusercontent.com/91914386/210704500-a724262f-e782-4e6c-9dfe-10edd68ddd28.png">

**Reply by tianxiewuhua** [reply]
> <img alt="Captura de pantalla 2023-01-04 a la(s) 22 53 29" width="792" src="https://user-images.githubusercontent.com/91914386/210704500-a724262f-e782-4e6c-9dfe-10edd68ddd28.png">

@jcdevsoft  this works for me！ thanks！！！

**Reply by ihu5** [reply]
guys you saved my life with serial option now it works

**Comment by andrew-ocallaghan-ie** [comment]
Hey folks, 

I'm running 14" 2021 Apple M1 Pro
I found two ways to get Kali installed and running.


1. Using Kali 2022.3 (current version) and following  [kalis official docs for utm install](https://www.kali.org/docs/virtualization/install-utm-guest-vm/). 
    - The trick is that use console display option for install then after install enable full graphics. This is explained in the docs.
    - I found this the easiest method and it also puts you on the **current** release.

2. Installing older version of Kali 2022.1 release as suggested above by @DylKhn. 
    - This worked and got me going in a pinch before I had read the kali docs.
    - A full list of kali releases can be found [here](https://old.kali.org/kali-images/). 
    - Select the [2022.1 release](https://old.kali.org/kali-images/kali-2021.1/) 
    - There's three types of arm64 installers. You probably want [default installer]([kali-linux-2022.3-installer-arm64.iso](https://old.kali.org/kali-images/kali-2022.3/kali-linux-2022.3-installer-arm64.iso)).
    - If you need a different installer the differences are explained [here](https://www.kali.org/docs/introduction/what-image-to-download/).

The steps are similar in both. You can use the [Kali UTM docs](https://www.kali.org/docs/virtualization/install-utm-guest-vm/)) to aid building your vm.

**Reply by ghost** [reply]
Hey andrew-ocallaghan-ie,
I too use a MacBook 14" 2021 Apple M1 Pro.
1 - Following the instructions [kalis official docs for umm install](https://www.kali.org/docs/virtualization/install-utm-guest-vm/)  the installation of Kali 2022.3 hangs on the screen (see image).
Do you have any idea what I can do to unblock the situation please?

2 - The installation works very well with Kali 2022.1

!![2022-12-07_17-42-47](https://user-images.githubusercontent.com/3705917/206239251-8ee29086-cef9-4586-a434-0252f2656b95.png)

**Comment by hvaandres** [comment]
The solution to this problem is the following:

1.- You will have to choose: The display option | Emulated Display Card - virtio-ramfb
2.- Add a Serial under Devices from the settings options of the VM
3.- You will get a screen asking you what type of software you want to install. Then, you will have to install Kali Linux without any software, you will have to uncheck all of the software that has been checked by default. 
4.- After you get a prompt window with "The installation is ready" You have to eject the ISO image
5.- Restart the machine, and login with your credentials
6.- Run the following commands:

```
sudo apt update
sudo apt install -y kali-desktop-kde
sudo update-alternatives --config x-session-manager
```

<img width="783" alt="Screenshot 2023-01-14 at 8 47 47 PM" src="https://user-images.githubusercontent.com/26829975/212521980-c7507d26-1eef-4253-9867-d3bcc12ef650.png">

<img width="886" alt="Screenshot 2023-01-14 at 8 50 56 PM" src="https://user-images.githubusercontent.com/26829975/212522032-9be29439-e7af-4f33-a98d-3064232217f3.png">

<img width="1093" alt="Screenshot 2023-01-14 at 8 56 04 PM" src="https://user-images.githubusercontent.com/26829975/212522154-30e5065d-5ea7-4553-818c-5524f4a6d0e1.png">


This will solve the black screen. I just downloaded the latest version of Kali and UTM, please ping me if you have any questions. If you would like to watch a video about this, you can check out this link [How to install kali linux through UTM](https://www.youtube.com/watch?v=C-Aa8_iB5Xs) OR [check out my blog about this issue](https://ithvaandres.medium.com/black-screen-while-installing-kali-linux-on-macs-m1-m2-through-utm-5d5939d93577)

**Reply by wgs45** [reply]
Thank you so much for the solution Mr @hvaandres, this works for me.

**Reply by s-en-o** [reply]
This works. What I did is change the display setting of Emulated Display Card from "virtio-ramfb-gl" to "virtio-ramfb" (notice there is no **_gl_** at the end). Also removing Serial from Devices setting still works fine (after finished the installation).

**Reply by noethig** [reply]
Thanks a lot, that also worked for me! :)

**Reply by chencxf12369** [reply]
This is really a quick and clear resolution share! Thanks!

**Comment by outbox3399** [comment]
@hvaandres Thank you, this worked for me. Thank you.

**Comment by ahmadshahab781** [comment]
https://youtu.be/9gXxQ0-wiNg

**Comment by ayushpattnaik** [comment]
#4983 solves this

**Reply by ahmadshahab781** [reply]
if You Use Utm virtual machine please download Kali Linux 2022.3 that will be Worked

**Comment by Fromtheoaks** [comment]
https://www.kali.org/docs/virtualization/install-utm-guest-vm/

this should help! :)

**Comment by earslap** [comment]
If your installation with the console mode (secondary small screen) freezes at random screen transitions with partial garbage updates, run the installation with "high contrast" mode (available in the initial menu) which seems to bypass the issue. could not find anything about this anywhere but this works around it. Hope it helps someone.
